pyatlan/model/core.py

```diff
@@ -275,17 +275,39 @@ def process_attributes_and_flush_cm(cls, asset):
         if not isinstance(asset, Asset):
             return asset
 
-        # Manually need to set these to "None" so that we can exclude
+        # Initialize set for attributes to exclude from serialization
+        exclude_attributes = set()
+        # Manually need to set these to "{}" so that we can exclude
         # them from the request playload when they're not set by the user
-        asset.remove_relationship_attributes = None
-        asset.append_relationship_attributes = None
+        asset.remove_relationship_attributes = {}
+        asset.append_relationship_attributes = {}
+        # Process relationship attributes and update exclusion set
         for attribute in asset.attributes:
-            asset = cls.process_relationship_attributes(asset, attribute)
-
-        # Flush custom metadata
+            exclude_attributes.update(
+                cls.process_relationship_attributes(asset, attribute)
+            )
+        # Determine relationship attributes to exclude
+        # https://docs.pydantic.dev/1.10/usage/exporting_models/#advanced-include-and-exclude
+        exclude_relationship_attributes = {
+            key: True
+            for key in [
+                "remove_relationship_attributes",
+                "append_relationship_attributes",
+            ]
+            if not getattr(asset, key)
+        }
+        if exclude_attributes:
```

Comment:
Regarding my previous comment I added the following line and all the integration and unit tests passed so it would seem that you only need to check the attributes in `asset.attributes.__fields_set__`:
`assert exclude_attributes.issubset(asset.attributes.__fields_set__)`

Reply:
Yes, it works!
Thanks.